graphql_query_derive/src/lib.rs

```diff
@@ -66,6 +66,26 @@ fn read_file(
     Ok(out)
 }
 
+fn implicitly_discover_query_name(struct_name: &str) -> Result<String, failure::Error> {
+    let mut out = "src/".to_owned();
+    out.push_str(&struct_name.to_string().to_snake_case());
+    out.push_str(".graphql");
+    Ok(out)
+}
+
+fn implicitly_discover_schema_name(cargo_manifest_dir: &str) -> Result<String, failure::Error> {
+    if ::std::path::Path::new(&cargo_manifest_dir).join("src/schema.json").exists() {
+        return Ok("src/schema.json".to_owned());
+    }
+    if ::std::path::Path::new(&cargo_manifest_dir).join("src/schema.graphql").exists() {
+        return Ok("src/schema.graphql".to_owned());
+    }
+    if ::std::path::Path::new(&cargo_manifest_dir).join("src/schema.gql").exists() {
+        return Ok("src/schema.gql".to_owned());
+    }
```

Comment:
I'm sure there's a much more elegant way of writing this, but I'm a newb so I'll gladly take suggestions.

Reply:
What about something like this?

```rust
fn find_schema_file(cargo_manifest_dir: &str) -> Result<String, failure::Error> {
    let default_paths = vec!["src/schema.json", "src/schema.graphql", "src/schema.gql"];

    default_paths
        .iter()
        .map(|path| Path::new(&cargo_manifest_dir).join(path))
        .find(|path| path.exists())
        .and_then(|path| path.to_owned())
        .ok_or_else(|| Err(format_err!("None of the following exist in src/: schema.json, schema.graphql, schema.gql")))
}
```

Reply:
That looks much nicer! Thanks

Reply:
+1 for this solution. I'm not sure but I think it can even be an array literal instead of `vec!`, so `let paths = &["src/schema.json", "src/schema.graphql", "src/schema.gql"];`, which would be a tiny bit more efficient but we really don't care in that context since it only runs once per query at compile-time.

Reply:
Yeah, I believe that should work as well. 👍